index.js

```diff
@@ -1,32 +1,34 @@
-const pad = n => `${n < 10 ? "0" : ""}${n}`;
+const dayjs = require("dayjs");
+const utcPlugin = require("dayjs/plugin/utc");
+
+dayjs.extend(utcPlugin);
+
 const getDateFor = ({ day = 1, month, year }) =>
-  new Date(`${year}-${pad(month)}-${pad(day)}T00:00:00.000Z`);
+  dayjs(`${year}-${month}-${day}`, "YYYY-MM-DD");
 
 const getNthDayOf = (n, day, month, year) => {
-  const firstOfMonth = getDateFor({ month, year });
+  let result = dayjs(getDateFor({ month, year })).day(day);
 
-  let dayOffset = firstOfMonth.getUTCDay() - day;
-  if (dayOffset > 0) {
-    dayOffset = 7 - dayOffset;
-  } else {
-    dayOffset = -dayOffset;
+  if (result.month() !== month - 1) {
```

Comment:
This is much more readable with dayjs! Nice. One thing I'd ask is for some comments explaining the logic around how `day()` works, because it wasn't immediately clear to me if it only went forward or if it would also go backwards. I want to help future devs understand what's happening.

```suggestion
  // dayjs.day(x) can return a time in the past (relative to the date being
  // operated on), because it returns a time from within the operand date's
  // current week. E.g.:
  //
  // date = July 1, 2021 # Thursday
  // dayjs(date).day(0)  # Get Sunday
  // # returns June 27, 2021
  if (result.month() !== month - 1) {
```